packages/react-batch/README.md

```diff
@@ -1,3 +1,58 @@
 # react-batch
 
-Exports [boxart](github.com/boxart/boxart)'s Batch and BatchFactory. If using more of boxart use boxart-batch instead. Tools for building html games with React.
+Exports [boxart](https://github.com/boxart/boxart)'s Batch and BatchFactory. If using more of boxart use boxart-batch instead. Tools for building html games with React.
+
+## usage
+
+`<Batch>` and `<BatchFactory>` make optimization assumptions on passed item arrays so that element creation and re-renders are reduced. If the items passed compared with the last props are through a shallow compare unchanged the underlying react virtual element isn't recreated (saving some time, object creation and garbage collection) and the object won't be re-rendered (saving further time, object creation, and garbage collection).
+
+Their use is to replace mapping an array of data to React elements.
+
+```js
+{items.map(item => <ItemComponent item={item} />)}
+```
+
+### Batch
+
+Mapping an array is replaced with a Batch component, taking the properties `items`, `itemKey`, `children` and `component`.
+
+- `items` is simply the array whose rendering it is helping to optimize.
+- `itemKey` is a required handle to produce a key to represent a member of the items array.
+- `children` is our handler for creating components for members of the items array.
+- `component` specifies the tag wrapping the rendered items. It defaults to `'div'`.
+
+```js
+<Batch items={items} itemKey={item => item.key}>{(
+  item => <ItemComponent item={item} />
+)}</Batch>
+```
+
+`children` in most React uses are a set of deeper React elements. Instead Batch like react-intl uses the property to produce the representations of members of the items array. This handle is called for each member of items when Batch is first created and everytime a member changes. Batch does not rebuild the rendered elements if `children` changes, so its best to not pass a different function to the same batch.
+
+`itemKey` is required to help make the best of `Batch`. Without `itemKey` Batch's implementation would have to recreate every element after any removed elements. With `itemKey` it can instead reuse the non-removed elements as they as stored by that key.
+
+#### Immutability
+
+Since items is expected to be treated immutably, `children` is only passed the item member of an array. It doesn't get the index or the containing array like `.map()` does. It is best to put all the relevant data for rendering the returned component from Batch's handler into the item in the array to start. This best avoids plenty of possible bugs and ensures the code understanbly re-renders when that item changes.
```

Comment:
Recommended to rephrase slightly:

> The array of `items` to be rendered is expected to be treated as immutable. Unlike `.map()`, which gets passed an index and the containing array, the `children` function is only passed the member `item` itself. It is therefore recommended that the `item`s in the `items` array contain any and all data needed to render and return the component from Batch's handler, to avoid possible bugs that might be encountered if that handler function depends on data from outer scopes.

Not wedded to that last sentence, I think it's an important thing to call out but I don't believe my phrasing of that last clause is a significant improvement.

I'd also say "re-renders in a predictable way" instead of "understandably", if you stick with the existing language.